### PR TITLE
Import wiki contents to a sphinx-doc website

### PR DESCRIPTION
**Pull request by TristanCacqueray**
This change imports the wiki contents (as [discussed](https://discourse.dhall-lang.org/t/contributing-to-the-wiki-documentation/107/8)) and use sphinx to render
an HTML website. The main features provided by sphinx I could not easily
implements in other website generators are:

* Table of content and navigation
* Linkable section heading with anchor
* Search

However sphinx (docutils) is best use with ReStructuredText as it is
arguably better suited for technical documentation. If using another markup
format is an option, then I vote for rST (and keep sphinx) or go with asciidoc.

### CODE REVIEW

**Comment by sjakobi** [comment]
Wow! Can you share some screenshots?

Regarding the markup: How close is Sphinx' Markdown dialect (Remarkdown?) to GitHub Flavored Markdown?

Regarding the idea of switching to rST or asciidoc – I think one advantage of sticking with Markdown is that many potential contributors will already be quite familiar with it, and I worry that using a different markup format might discourage them.

**Comment by TristanCacqueray** [comment]
Here are some screenshot, using the default theme:

main-page
reference-page

The only issue so far with GitHub Flavored Markdown (gfm) is the link format `[[Title|Doc]]` that need to be adapted to `[Title](file-path)`.

Agreed markdown is the most popular, but rST is easy to learn too. FWIW I don't mind either way.

**Comment by TristanCacqueray** [comment]
I forgot to mention, the important files are the `README`, `conf.py` and `index.md` files. The rest are mostly imported from the wiki.

**Comment by philandstuff** [comment]
I have previously used Sphinx and rst and I’m happy with them, despite the ubiquity of markdown

**Comment by Gabriel439** [comment]
I pushed the Nix-related changes to this branch.  Specifically:

* You can now build the docs by running:

  ```bash
  nix build --file ./release.nix docs
  ```

* The NixOps deploy now serves `docs.dhall-lang.org`

I also went ahead and deployed this branch to https://docs.dhall-lang.org/ so you can begin to browse the documentation! (and once this is merged I'll update dhall-lang.org to link to docs.dhall-lang.org)

File: docs/howtos/How-to-add-a-new-built-in-function.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# How to add a new built in function
```
Comment (sjakobi) [inline docs/howtos/How-to-add-a-new-built-in-function.md]:
```suggestion
# How to add a new built-in function
```

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Dhall Lang Documentation
```
Comment (Gabriel439) [inline docs/index.md]:
```suggestion
Dhall Documentation
```

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Sphinx project configuration
+project = 'Dhall Lang'
```
Comment (Gabriel439) [inline docs/conf.py]:
```suggestion
project = 'Dhall'
```

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+========================
+
+Welcome to the Dhall Lang documentation.
```
Comment (Gabriel439) [inline docs/index.md]:
```suggestion
Welcome to the Dhall documentation!
```

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+``` important:: This is a work in progress
```
Comment (Gabriel439) [inline docs/index.md]: Rather than saying that this is a work in progress, I would instead provide instructions here for how to contribute

**Comment by TristanCacqueray** [comment]
Arg, I messed this up by incorrect rebase.

**Comment by Gabriel439** [comment]
@TristanCacqueray: As a general rule of thumb you don't ever need to rebase.  Just add commits to the branch

We squash merge the final result onto `master` when we're done so the result is a single commit

**Comment by TristanCacqueray** [comment]
@Gabriel439 I'm not used to GitHub workflow, I thought I had to pull the new commits before pushing the follow-up. Thus I did `git fetch origin refs/pull/826/head` and `git checkout FETCH_HEAD` to get the new commits, and after pushing my local branch it created a `merge commit` because of 5b5fa72e. Then trying to squash the merge commit I must have push the wrong tip, resulting in this PR to close prematurely.

Thanks a lot for the feedback, I'm glad you like the result. It's really nice to see this work already live at docs.dhall-lang.org!

**Comment by TristanCacqueray** [comment]
@philandstuff So far the markdown compat seems usable, and we could switch to rst iteratively if needed in the future.

**Comment by SiriusStarr** [comment]
I personally vote for markdown over rST.  I personally can't stand rST, haha, but whatever it ends up being I'm sure people can work with.

**Pull request closed without merging**